### DISCUSSION TITLE
params ignored when sending POST request

### DISCUSSION THREAD

**Discussion by gasta-qf** [mlflow/mlflow#13004] [Q&A]
Hi, guys.

I have the following PythonModel as an extension of the statsmodel.

```python
class ModelWrapperOLS(mlflow.pyfunc.PythonModel):
    def __init__(self, uri):
        self.model = None
        self.uri = uri

    def load_context(self, context):
        self.model = mlflow.statsmodels.load_model(self.uri)

    def predict(
        self, context, model_input, params=None
    ):
        params = params or {"predicted_variable": "A"}
        predicted_variable = params.get("predicted_variable")

        if predicted_variable == "A":
            return 0
        elif predicted_variable == "B":
            return 1
        else:
            raise ValueError(
                f"The predicted variable '{predicted_variable}' is not supported."
            )
```

When I launch the model locally on my machine using `mlflow models serve ... -p 5100` and send a POST request with current data 
``` JSON
 {
    "dataframe_split": {
        "columns": ["0"],
        "data": [[14]]
    },
    "params" : {"predicted_variable" : "B"}
}
``` 
and I get **1** as a return which is the desired behaviour.

However, when I first build an image (using mlflow models build-docker) and run that image in the local docker container **or** deploy it to the k8s, I always get a response for default option (`predicted_variable = A`), regardless of what I put in params. As if the params option is completely ignored when launched on the container. 

So as a summary: params argument works as intended when launched on the machine using mlflow models serve and is ignored if launched on the container. 

Thanks in advance!

EDIT:

I discovered what the issue was. The reason it work locally but did not work in the container, is because I launch it with mlserver disabled locally and with it being enabled on the container. In my understanding it should not make any difference, however it does (which might be a bug?). Now when I deployed it with default flask, it work everywhere as intended

**Comment by bikeholik** [comment]
it seems that `mlserver_mlflow/runtime.py` doesn't pass any parameters extracted from the request to the prediction. it just parses input data
